cli/console.ts

```diff
@@ -45,6 +45,100 @@ export function question(questionText: string) {
+ * non-TTY (i.e. test) environment and DATAFORM_CLI_TEST_INPUTS is set,
+ * it consumes inputs from the environment variable.
+ * @param questionText The question text to display.
+ * @returns Users's input or test input.
+ */
+export function interactiveQuestion(questionText: string): string {
+  // If running in a non-TTY environment and test inputs are provided
+  if (!process.stdin.isTTY && process.env.DATAFORM_CLI_TEST_INPUTS !== undefined) {
+    const testInput = getTestInput(questionText);
+    if (testInput !== undefined) {
+      print(`${questionText} ${testInput}\n`); // Echo the test input for clarity in logs
+      return testInput;
+    }
+    // If we are in a non-TTY  (i.e. test) environment and not enough test inputs were provided
+    throw new Error("Not enough test inputs provided in DATAFORM_CLI_TEST_INPUTS.");
+  }
+
+  // If running in TTY environment or if testInputs are not available in non-TTY
+  print(`${questionText} `);
+  return readlineSync.question("");
+}
+
+/**
+ * Represents the state for managing test inputs provided via environment
+ * variables. This state uses a map to associate question texts with answers.
+ */
+interface ITestInputsState {
+  /**
+   * A map where keys are question strings and values are the pre-defined answers.
+   */
+  inputs: Map<string, string>;
+}
+let testInputsState: ITestInputsState | undefined;
+
+/**
+ * Helper function to enable testing interactive CLI. Initializes
+ * testInputsState by parsing the DATAFORM_CLI_TEST_INPUTS environment variable.
+ * This function ensures the parsing happens only once.
+ */
+function ensureTestInputsInitialized(): void {
+  if (testInputsState !== undefined) {
+    return; // Already initialized
+  }
+
+  const envVar = process.env.DATAFORM_CLI_TEST_INPUTS;
+  if (!envVar) {
+    return; // Environment variable not set
+  }
+
+  try {
+    const parsed = JSON.parse(envVar);
+    // Expect a JSON object where keys are questions and values are answers.
+    if (typeof parsed === 'object' && parsed !== null && !Array.isArray(parsed)) {
+      testInputsState = {
+        inputs: new Map<string, string>(Object.entries(parsed).map(([key, value]) => [key, String(value)])),
+      };
+      printError("Using DATAFORM_CLI_TEST_INPUTS map for console prompts");
+    } else {
+      printError(`Failed to parse DATAFORM_CLI_TEST_INPUTS: Expected a JSON object, but got ${typeof parsed}.`);
+      testInputsState = undefined;
+    }
+  } catch (e) {
+    printError("Failed to parse DATAFORM_CLI_TEST_INPUTS:", e);
+    testInputsState = undefined;
+  }
+}
+
+/**
+ * Helper function to enable testing interactive CLI. Retrieves testInput from
+ * the testInputsState map using the question text as the key.
+ * @param questionText The exact question text displayed to the user.
+ * @returns Test input, or undefined if no test inputs are available for this question.
+ */
+function getTestInput(questionText: string): string | undefined {
```

Comment:
Could we make it simple and here check if there is a process.env.DATAFORM_CLI_TEST_INPUTS provided, than try to parse it. If it fails - throw, if parsed correctly - read from map by key.